graylog2-server/src/main/java/org/graylog/events/processor/aggregation/AggregationEventProcessorConfig.java

```diff
@@ -194,37 +206,60 @@ public ValidationResult validate() {
         if (!series().isEmpty() && isConditionsEmpty()) {
             validationResult.addError(FIELD_CONDITIONS, "Aggregation with series must also contain conditions");
         }
+        return validationResult;
+    }
+
+    @Override
+    public ValidationResult validate(@Nullable EventProcessorConfig oldEventProcessorConfig,
+                                     EventDefinitionConfiguration eventDefinitionConfiguration) {
+        final ValidationResult validationResult = new ValidationResult();
+        AggregationEventProcessorConfig oldConfig = (AggregationEventProcessorConfig) oldEventProcessorConfig;
+
+        if (oldConfig == null) {
+            if (eventLimit() <= 0) {
+                validationResult.addError(FIELD_EVENT_LIMIT, "Event limit must be greater than 0.");
+            }
+        } else {
+            if (oldConfig.eventLimit() != 0 && eventLimit() <= 0) {
```

Comment:
If someone changes his `oldConfig` from aggregation to filter, we can also enforce the new event limit.
I guess we can append `oldConfig.series().isEmpty() &&` to the query to check this